lib/puppet/face/config.rb

```diff
@@ -106,17 +130,24 @@
     EOT
 
     when_invoked do |name, value, options|
+
+      options[:section] = options[:section].to_s # If value was left as default - set to default string
+
       if name == 'environment' && options[:section] == 'main'
-        Puppet.warning _(<<-EOM).chomp
-The environment should be set in either the `[user]`, `[agent]`, or `[master]`
-section. Variables set in the `[agent]` section are used when running
-`puppet agent`. Variables set in the `[user]` section are used when running
-various other puppet subcommands, like `puppet apply` and `puppet module`; these
-require the defined environment directory to exist locally. Set the config
-section by using the `--section` flag. For example,
-`puppet config --section user set environment foo`. For more information, see
-https://puppet.com/docs/puppet/latest/configuration.html#environment
-        EOM
+        messages = []
+        #TRANSLATORS `[user]`, `[agent]`, and `[master]` are section names and should not be translated
+        messages << _("The environment should be set in either the `[user]`, `[agent]`, or `[master]` section.")
+        #TRANSLATORS `[agent]` is a section name and `puppet agent` is a command line example and should not be translated
+        messages << _("Variables set in the `[agent]` section are used when running `puppet agent`.")
+        #TRANSLATORS `[user]` is a section name and `puppet apply` and `puppet module` are command line examples and should not be translated
+        messages << _("Variables set in the `[user]` section are used when running various other puppet subcommands,\n"\
+          "  like `puppet apply` and `puppet module`;\n  these require the defined environment directory to exist locally.")
```

Comment:
the 2 leading spaces will make english show the indention, but likely get lost in translated languages depending on whether the translators add the spaces or not. I'd probably just make it one long line and let it wrap.  (shrug)